apsi-enrollment-be/src/main/java/edu/pw/apsienrollment/common/db/SearchQueryParser.java

```diff
@@ -0,0 +1,21 @@
+package edu.pw.apsienrollment.common.db;
+
+import lombok.NonNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SearchQueryParser {
+    public static Collection<SearchCriteria> parse(@NonNull String searchQuery) {
+        Pattern pattern = Pattern.compile("([\\w.]+?)(:|>|>=|<|<=)((\\w|:|-|;|\\p{Space})+?),", Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
`((\\w|:|-|;|\\p{Space})+?)` inner (4th) search group is unused as search group; that fragment could be replaced with:
`([\\w:-;\\ ]+?)`

Reply:
I suggest splitting this string into a number of named constants to improve readability

Reply:
Splitting regex into substrings in not a good practise, cause it represent single patern to test/match against. Splitting it into substrings would make it way less readable and testable via external regex testing tools.